cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/analyzer/goals/GoalUtilsTest.java

```diff
@@ -0,0 +1,91 @@
+
+    // Verify 3: Not low utilization and compute balance threshold lower bound
+    verifyComputingResourceUtilizationBalanceThreshold(0.3,
+                                                       resource,
+                                                       ResourceDistributionGoal.BALANCE_MARGIN,
+                                                       true,
+                                                       true,
+                                                       0.2,
+                                                       1.3,
+                                                       1.2,
+                                                       0.3 * (1 - ((1.3 * 1.2) - 1) * ResourceDistributionGoal.BALANCE_MARGIN));
+
+    // Verify 4: Not low utilization and compute balance threshold upper bound
+    verifyComputingResourceUtilizationBalanceThreshold(0.3,
+                                                       resource,
+                                                       ResourceDistributionGoal.BALANCE_MARGIN,
+                                                       false,
+                                                       true,
+                                                       0.2,
+                                                       1.3,
+                                                       1.2,
+                                                       0.3 * (1 + ((1.3 * 1.2) - 1) * ResourceDistributionGoal.BALANCE_MARGIN));
+
+  }
+
+  private void verifyComputingResourceUtilizationBalanceThreshold(double avgUtilizationPercentage,
+                                                                  Resource resource,
+                                                                  double balanceMargin,
+                                                                  boolean isLowerThreshold,
+                                                                  boolean isTriggeredByGoalViolation,
+                                                                  double lowUtilizationThreshold,
+                                                                  double resourceBalancePercentage,
+                                                                  double goalViolationDistributionThresholdMultiplier,
+                                                                  double expectedComputedBalanceThreshold) {
+
+    BalancingConstraint mockBalanceConstraint = EasyMock.mock(BalancingConstraint.class);
+    EasyMock.expect(mockBalanceConstraint.lowUtilizationThreshold(resource)).andReturn(lowUtilizationThreshold).anyTimes();
+    EasyMock.expect(mockBalanceConstraint.resourceBalancePercentage(resource)).andReturn(resourceBalancePercentage).anyTimes();
+    EasyMock.expect(mockBalanceConstraint.goalViolationDistributionThresholdMultiplier()).
+        andReturn(goalViolationDistributionThresholdMultiplier).anyTimes();
+
+    EasyMock.replay(mockBalanceConstraint);
```

Comment:
Is EasyMock.verify(XXX) missing intentionally?

Reply:
It was missed. I will add it back